inbm/dispatcher-agent/tests/unit/sota/test_sota.py

```diff
@@ -130,7 +130,8 @@ def test_run_raises(self, mock_reboot, mock_rollback_and_delete_snap, mock_print
                 mock_rollback_and_delete_snap.assert_called_once()
             mock_reboot.assert_called_once()
         except SotaError as e:
-            self.assertEqual(str(e), "SOTA cache directory cannot be created")
+            assert (str(e) == "SOTA cache directory cannot be created" or
+                str(e) == "Final result in SOTA execution: SOTA fail")
```

Comment:
**Suggestion:** Replace the assertion with self.assertIn to check if the exception message is one of the expected messages. This is more idiomatic in unittest and provides better error messages on failure. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            except SotaError as e:
                self.assertIn(str(e), ["SOTA cache directory cannot be created",
                                       "Final result in SOTA execution: SOTA fail"])
```